internal/ptx/obfs4_test.go

```diff
@@ -0,0 +1,138 @@
+package ptx
+
+import (
+	"context"
+	"errors"
+	"net"
+	"strings"
+	"testing"
+
+	"github.com/ooni/probe-cli/v3/internal/atomicx"
+	"github.com/ooni/probe-cli/v3/internal/engine/netx/mockablex"
+)
+
+func TestOBFS4DialerWorks(t *testing.T) {
+	// This test is 0.3 seconds in my machine, so it's ~fine
+	// to run it even when we're in short mode
+	o4d := DefaultTestingOBFS4Bridge()
+	conn, err := o4d.DialContext(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+	if conn == nil {
+		t.Fatal("expected non-nil conn here")
+	}
+	if o4d.Name() != "obfs4" {
+		t.Fatal("unexpected value returned by Name")
+	}
+	bridgearg := o4d.AsBridgeArgument()
+	expectedbridge := "obfs4 109.105.109.165:10527 8DFCD8FB3285E855F5A55EDDA35696C743ABFC4E cert=Bvg/itxeL4TWKLP6N1MaQzSOC6tcRIBv6q57DYAZc3b2AzuM+/TfB7mqTFEfXILCjEwzVA iat-mode=1"
```

Comment:
Again probably use bridges from tor browser in here.

Reply:
thanks